core/transport/src/grpc/meta_master.proto

```diff
@@ -112,6 +112,20 @@ message GetMetricsPResponse {
   map<string, MetricValue> metrics = 1;
 }
 
+message SnapshotPOptions {}
+
+/**
+ * The snapshot response has three cases:
+ * 1. failed to trigger (triggered=false and message explains why snapshot is not triggered)
+ * 2. triggered but failed (triggered=false, succeed=false and message explains why snapshot is failed)
+ * 3. succeed (triggered=true, succeed=true)
+ */
+message SnapshotPResponse {
+ optional bool triggered = 1;
+ optional bool succeed = 2;
+ optional string message = 3;
```

Comment:
usually we use exceptions for this sort of thing. If the rpc fails to trigger a snapshot, we throw an exception saying why it failed to trigger. If the rpc starts a snapshot but the snapshot fails, we throw an exception saying why the snapshot failed. If we don't throw anything, that means it succeeded.

Reply:
Yeah, will use exceptions instead of message here to indication the rpc fails.
But Snapshot has one more circumstances other than snapshotted or failed: the snapshot is not triggered. 

Is it OK that we only responsible for giving the possibility of taking a snapshot, but not guarantee snapshotted in the Snapshot CLI? Giving console message like: successfully send the snapshot signal.

Reply:
I think we can throw an exception with a relevant message if we fail to checkpoint because the journal doesn't want to be checkpointed, e.g. Copycat decides not to snapshot, or in the UFS journal no entries have been written since the last checkpoint. I don't see a need for the client side to differentiate between "Failed because the journal refused" and "Failed for some other reason". Either way I think we'll just forward the error message to the client

Reply:
Make sense. Now I will also throw exceptions when not be able to trigger snapshot. Only return results when snapshot is triggered and succeed.